prow/plugins/trigger/trigger.go

```diff
@@ -260,3 +258,33 @@ func skipRequested(c Client, pr *github.PullRequest, skippedJobs []config.Presub
 	}
 	return errorutil.NewAggregate(errors...)
 }
+
+func getPresubmits(c Client, pr *github.PullRequest) (string, []Presubmit, error) {
+	inRepoConfigEnabled := c.Config.InRepoConfigConfiguration(org, repo).Enabled
+	// We do the status context handling here instead of in the  PluginAgent to avoid
+	// updating it once to Pending and then Success/Failed per plugin that accesses the
+	// Presubmit config
+	if inRepoConfigEnabled {
+		// Create GitHub status context "prow-config-parsing" with status "Pending"
+	}
+	baseSHA, presubmits, err := c.GetPresubmits(pr)
+	if err != nil {
+		if inRepoConfigEnabled {
+			// Update GitHub status context "prow-config-parsing" with status "Failed"
+		}
+		return "", nil, err
+	}
+
+	if inRepoConfigEnabled {
+		// * There is exactly one case where we need to retire a context: If it was removed
+		//   on the base branch but is failed on our PR. We handle it by:
+		//   * Iterating over all existing contexts that do not have a "Success" status
+		//   * Ignore "tide" and "prow-config-parsing" contexts
+		//   * Ignore contexts whose `target_url` does not start with c.Config.GetJobUrlPrefix()
+		//   * check if found context is in presubmits if not, retire it
+		//   TODO: Can we use this approach everywhere and use it to replace the status-reconciler?
```

Comment:
boooooooo

Reply:
Doing it explicitly when we change config seems much saner than by default on every PR interaction. What's the impact of listing statuses on ... every webhook?

Reply:
One additional token per pull request event. If there is a third party updating statuses, also up to an additional token per comment that triggers trigger.